python/cudf/cudf/utils/utils.py

```diff
@@ -489,39 +489,64 @@ def get_appropriate_dispatched_func(
         if cupy_func is func:
             return NotImplemented
 
-        cupy_compatible_args = get_cupy_compatible_args(args)
-        cupy_output = cupy_func(*cupy_compatible_args, **kwargs)
-        return cast_to_appropriate_cudf_type(cupy_output)
+        cupy_compatible_args, index = _get_cupy_compatible_args_index(args)
+        if cupy_compatible_args:
+            cupy_output = cupy_func(*cupy_compatible_args, **kwargs)
+            return _cast_to_appropriate_cudf_type(cupy_output, index)
+        else:
+            return NotImplemented
     else:
         return NotImplemented
 
 
-def cast_to_appropriate_cudf_type(val):
+def _cast_to_appropriate_cudf_type(val, index=None):
     # TODO Handle scalar
     if val.ndim == 0:
         return cudf.Scalar(val).value
     # 1D array
     elif (val.ndim == 1) or (val.ndim == 2 and val.shape[1] == 1):
-        return cudf.Series(val)
+        if (index is None) or (len(index) == len(val)):
+            return cudf.Series(val, index=index)
+        else:
+            # if index is not None and is of a different length
+            # than the index cupy dispatching behaviour is undefined
+            return NotImplemented
```

Comment:
I think you can just dump to the outer return here

Reply:
Resolved by https://github.com/rapidsai/cudf/pull/6839/commits/665234136db75eb1b1024e125bff853df6cf27a1